web/src/pages/insights/SummaryTable.tsx

```diff
@@ -0,0 +1,294 @@
+import React, { ReactNode, useState } from 'react'
+import { Table as SUITable } from 'semantic-ui-react'
+import Table from '../../shared/components/Table'
+import { ThemeContext } from '../../shared/components/ThemeProvider'
+import HtmlTooltip from '../../shared/utilities/htmlTooltip'
+import { ProjectInsightsSummary } from '../../sm-api'
+import { ColumnKey, HeaderCell, summaryTableHeaderCellConfigs } from './HeaderCell'
+import { FooterCell, footerCellConfigs } from './SummaryTableFooterCell'
+
+interface SummaryTableProps {
+    filteredData: ProjectInsightsSummary[]
+    handleSelectionChange: (columnName: ColumnKey, selectedOptions: string[]) => void
+    getUniqueOptionsForColumn: (key: ColumnKey) => string[]
+    getSelectedOptionsForColumn: (key: ColumnKey) => string[]
+}
+
+interface PercentageCellProps {
+    percentage: number
+    tooltipContent: ReactNode
+    isDarkMode: boolean
+    getPercentageColor: (percentage: number, isDarkMode: boolean) => string
+}
+
+const PercentageCell: React.FC<PercentageCellProps> = ({
+    percentage,
+    tooltipContent,
+    isDarkMode,
+    getPercentageColor,
+}) => (
+    <SUITable.Cell
+        className="percentage-cell"
+        style={{ backgroundColor: getPercentageColor(percentage, isDarkMode) }}
+    >
+        <HtmlTooltip title={tooltipContent}>
+            <div>{percentage.toFixed(2)}%</div>
+        </HtmlTooltip>
+    </SUITable.Cell>
+)
+
+function calculatePercentage(count: number, total: number): number {
+    return count > 0 ? (count / total) * 100 : 0
+}
+
+function getPercentageColor(percentage: number, isDarkMode: boolean) {
+    const hue = (percentage / 100) * 120 // Convert percentage to hue value (0-120)
+    const saturation = isDarkMode ? '100%' : '90%' // Set saturation based on mode
+    const lightness = isDarkMode ? '25%' : '75%' // Set lightness based on mode
+
+    return `hsl(${hue}, ${saturation}, ${lightness})`
+}
+
+const getRowClassName = (sequencingType: string) => {
+    switch (sequencingType) {
+        case 'exome':
+            return 'exome-row'
+        case 'genome':
+            return 'genome-row'
+        default:
+            return 'rna-row'
+    }
+}
+
+const getSummaryCell = (
+    summary: ProjectInsightsSummary,
+    key: ColumnKey,
+    isDarkMode: boolean
+): React.ReactNode => {
+    // What goes into each cell of the table is determined by the column header or 'key'
+    // The percentage calculation cells are handled differently from the other cells
+    const percentageAligned =
+        summary.total_sequencing_groups > 0
+            ? (summary.total_crams / summary.total_sequencing_groups) * 100
+            : 0
+    const alignedCellTooltip = (
```

Comment:
This is great to have it pulled out, you'd usually go a step further and make this it's own tiny component

Reply:
Pulled each of the "percentage" cells into their own individual components, e.g. `GetAlignedPercentageCell`, `GetAnnotatedDatasetPercentageCell`, etc. 

Now, the `GetSummaryCell` component is a lot simpler and to the point.
```tsx
const GetSummaryCell: React.FC<{
    summary: ProjectInsightsSummary
    columnKey: ColumnKey
    isDarkMode: boolean
}> = ({ summary, columnKey, isDarkMode }) => {
    switch (columnKey) {
        case 'aligned_percentage':
            return <GetAlignedPercentageCell summary={summary} isDarkMode={isDarkMode} />
        case 'annotated_dataset_percentage':
            return <GetAnnotatedDatasetPercentageCell summary={summary} isDarkMode={isDarkMode} />
        case 'snv_index_percentage':
            return <GetSnvIndexPercentageCell summary={summary} isDarkMode={isDarkMode} />
        case 'sv_index_percentage':
            return <GetSvIndexPercentageCell summary={summary} isDarkMode={isDarkMode} />
        default:
            return (
                <SUITable.Cell className="table-cell">
                    {summary[columnKey as keyof ProjectInsightsSummary]}
                </SUITable.Cell>
            )
    }
}
```